### ISSUE TITLE
Error calculating display info on filtering with a model

### ISSUE THREAD

**Issue by darksciencebase** [metabase/metabase#36400]
> This seems MBQL-related. In the repro I:
> 1. Went to [the model](https://stats.metabase.com/model/14583)
> 2. Filtered in chill mode (which works)
> 3. Opened notebook mode
> 4. Removed the filter
> 5. Opened the filter UI again
> 6. It broke

[Slack Message](https://metaboat.slack.com/archives/C05MPF0TM3L/p1701771677901339?thread_ts=1701771677.901339&cid=C05MPF0TM3L) (there's a loom of the repro in the thread)

```
n0.gi {message: "Error calculating display info for column-group: Cannot read properties of null (reading 'replace')", data: n0.h, cause: TypeError: Cannot read properties of null (reading 'replace')
    at a (https://stats.metabase.com/…, name: 'Error', description: undefined, …}
iJ @ react-dom.production.min.js:209
n.payload @ react-dom.production.min.js:227
oO @ react-dom.production.min.js:129
iI @ react-dom.production.min.js:179
l @ react-dom.production.min.js:263
aH @ react-dom.production.min.js:246
(anonymous) @ react-dom.production.min.js:246
aW @ react-dom.production.min.js:246
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
on @ react-dom.production.min.js:122
oi @ react-dom.production.min.js:123
oo @ react-dom.production.min.js:122
Y @ react-dom.production.min.js:287
tU @ react-dom.production.min.js:68
```

**Comment by bshepherdson** [comment]
I'll take a look at this. It sounds like it might be the same cause as #37067, which has a fixed out for review #37536. Once that is in stats it'll be easy to try again to repro it.

**Comment by bshepherdson** [comment]
Actually I can't reproduce this.

Looking more closely at the steps and outcome, I suspect that this was fixed by removing empty stages (#37311). With a SQL model, creating the filter adds a new stage, and then removing the filter empties the stage. Then I guess something broke when trying to open the Filters UI again.

Closing as no repro.

**Comment by bshepherdson** [comment]
Ah, I can repro this. I'm looking into it.

**Comment by bshepherdson** [comment]
The immediate error is coming from trying to regex-match against the `:display-name` of a field, but it's `nil` rather than a string. The `:display-name` is nil I think because we don't have the target field in the metadata yet.

The columns for which it fails are mapped in from another table, and I guess if you land directly on the model and start adding and removing filters it might not have the metadata yet. I'm still investigating. We can certainly fix the code to be more robust against `nil`s here, but it would still be displaying the filterable columns badly.

I'm still debugging and trying to find the root problem, whether metadata or otherwise.

**Comment by bshepherdson** [comment]
Digging in makes it look like this might also be solved by fixing how implicit joins are handled with SQL models, see #37536 .

**Comment by bshepherdson** [comment]
#37536 is submitted; I'll check this tomorrow to see if it's fixed on stats.

**Comment by bshepherdson** [comment]
This was fixed by #37771.
